clients/python-apiserver-client/python_apiserver_client/params/cluster.py

```diff
@@ -0,0 +1,193 @@
+    It provides APIs to create, stringify, convert to dict and json.
+
+    Methods:
+    - Create cluster spec from: gets the following parameters:
+        head_group_spec - required, specification of the head node
+        worker_group_spec - optional, list of worker group specs
+    - to_string() -> str: convert toleration to string for printing
+    - to_dict() -> dict[str, any] convert to dict
+    """
+    def __init__(self, head_node: HeadNodeSpec, worker_groups: list[WorkerNodeSpec] = None) -> None:
+        self.head_node = head_node
+        self.worker_groups = worker_groups
+
+    def to_string(self) -> str:
+        val = f"head_group_spec: {self.head_node.to_string()}"
+        if self.worker_groups is not None:
+            val += "\nworker groups: "
+            for w in self.worker_groups:
+                val += f"\nworker_group_spec = {w.to_string()}]"
+        return val
+
+    def to_dict(self) -> dict[str, any]:
+        dst = {"headGroupSpec": self.head_node.to_dict()}
+        if self.worker_groups is not None:
+            dst["workerGroupSpec"] = [w.to_dict() for w in self.worker_groups]
+        return dst
+
+
+class ClusterEvent:
+    """
+    Cluster event is used to define events emitted during cluster creation.
+    It provides APIs to create and stringify. Its output only data, so we do not need to implement to_dict
+
+    Methods:
+    - Create templates: gets the following parameters:
+        id - unique Event Id
+        name - human readable event name
+        created_at - event creation time
+        first_timestamp - first time the event occur
+        last_timestamp - last time the event occur
+        reason - reason for the transition into the object's current status
+        message - human-readable description of the status of this operation
+        type - type of this event (Normal, Warning), new types could be added in the future
+        count - number of times this event has occurred
+    """
+    def __init__(self, dst: dict[str, any]) -> None:
+        self.id = dst.get("id", "")
+        self.name = dst.get("name", "")
+        self.created_at = dst.get("created_at", "")
+        self.first_timestamp = dst.get("first_timestamp", "")
+        self.last_timestamp = dst.get("last_timestamp", "")
+        self.reason = dst.get("reason", "")
+        self.message = dst.get("message", "")
+        self.type = dst.get("type", "")
+        self.count = dst.get("count", "0")
+
+    def to_string(self) -> str:
+        return (f"id = {self.id}, name = {self.name}, created_at = {self.created_at}, "
+                f"first_timestamp = {self.first_timestamp}, last_timestamp = {self.last_timestamp},"
+                f"reason = {self.reason}, message = {self.message}, type = {self.type}, count = {self.count}")
+
+
+class Cluster:
+    """
+    Cluster is used to define Ray cluster.
+    It provides APIs to create, stringify, convert to dict and json.
+
+    Methods:
```

Comment:
Same comments as above about wording on these two lines.

Reply:
Actually, here you aren't using key-value pairs (or a dictionary). Can you make these constructors consistent?